include/envoy/stats/stats.h

```diff
@@ -408,25 +408,33 @@ typedef std::unique_ptr<StoreRoot> StoreRootPtr;
 struct RawStatData;
 
 /**
- * Abstract interface for allocating a RawStatData.
+ * Abstract interface for allocating statistics. Alternate
+ * implementations can be created utilizing a single fixed-size block
+ * suitable for shared-memory, or in the heap, allowing for pointers
+ * and sharing of substrings, with an opportunity for reduced memory
+ * consumption.
  */
-class RawStatDataAllocator {
+class StatDataAllocator {
```

Comment:
It seems like this is a general stat factory class that determines the `impl` for each type of stat (and how it is allocated).  If it is able to choose the representation for gauges and counters, why not also allow it to make histograms?

Reply:
added TODO -- making a parallel path for histograms would require moving the histogram impl out of thread_local_store and into stats_impl, which seems OK to me, but not in this PR.  @ramaraochavali @mattklein123  WDYT?

Reply:
@jmarantz originally I have added it to Stats_impl only. But later moved to thread_local_store based on @mattklein123 suggestion. I think it is ok to bring it back to stats impl.